src/Swarm/Util.hs

```diff
@@ -213,26 +215,31 @@ getDataDirSafe p = do
 -- | Get file from swarm data directory.
 --
 -- See the note in 'getDataDirSafe'.
-getDataFileNameSafe :: FilePath -> IO (Maybe FilePath)
+getDataFileNameSafe ::
+  MonadIO m =>
+  FilePath ->
+  ExceptT PathLoadFailure m FilePath
 getDataFileNameSafe name = do
-  dir <- getDataDirSafe "."
-  case dir of
-    Nothing -> return Nothing
-    Just d -> do
-      let fp = d </> name
-      fe <- doesFileExist fp
-      return $ if fe then Just fp else Nothing
+  d <- ExceptT $ getDataDirSafe "."
+  let fp = d </> name
+  fe <- liftIO $ doesFileExist fp
+  ExceptT $
+    return $
+      if fe
+        then Right fp
+        else Left $ PathLoadFailure fp $ DoesNotExist File
```

Comment:
`PathLoadFailure` is used independently here.

Reply:
OK, since Util is the lowest library, you would have to move this failure type here.

Or do not use the type here.

_Everything_ now depends on it. Please try to imagine the dependency graph you are creating.